src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py

```diff
@@ -205,23 +220,35 @@ def get_token(server, resource, scope):  # pylint: disable=unused-argument
 def data_plane_azure_keyvault_administration_backup_client(cli_ctx, command_args):
     from azure.keyvault.administration import KeyVaultBackupClient
 
-    version = str(get_api_version(cli_ctx, ResourceType.DATA_KEYVAULT_ADMINISTRATION_BACKUP))
-    profile = Profile(cli_ctx=cli_ctx)
-    credential, _, _ = profile.get_login_credentials(resource='https://managedhsm.azure.net')
-    vault_url = \
-        command_args.get('hsm_name', None) or \
-        command_args.get('vault_base_url', None) or \
-        command_args.get('identifier', None)
-    if not vault_url:
-        raise RequiredArgumentMissingError('Please specify --hsm-name or --id')
+    vault_url, credential, version = _prepare_data_plane_azure_keyvault_client(
+        cli_ctx, command_args, ResourceType.DATA_KEYVAULT_ADMINISTRATION_BACKUP)
     return KeyVaultBackupClient(
         vault_url=vault_url, credential=credential, api_version=version)
 
 
 def data_plane_azure_keyvault_administration_access_control_client(cli_ctx, command_args):
     from azure.keyvault.administration import KeyVaultAccessControlClient
 
-    version = str(get_api_version(cli_ctx, ResourceType.DATA_KEYVAULT_ADMINISTRATION_ACCESS_CONTROL))
+    vault_url, credential, version = _prepare_data_plane_azure_keyvault_client(
+        cli_ctx, command_args, ResourceType.DATA_KEYVAULT_ADMINISTRATION_ACCESS_CONTROL)
+    return KeyVaultAccessControlClient(
+        vault_url=vault_url, credential=credential, api_version=version)
+
+
+def data_plane_azure_keyvault_key_client(cli_ctx, command_args):
+    from azure.keyvault.keys import KeyClient
+
+    vault_url, credential, version = _prepare_data_plane_azure_keyvault_client(
+        cli_ctx, command_args, ResourceType.DATA_KEYVAULT_KEYS)
+    command_args.pop('hsm_name', None)
+    command_args.pop('vault_base_url', None)
+    command_args.pop('identifier', None)
+    return KeyClient(
+        vault_url=vault_url, credential=credential, api_version=version)
+
+
+def _prepare_data_plane_azure_keyvault_client(cli_ctx, command_args, resource_type):
+    version = str(get_client_api_version(cli_ctx, resource_type))
     profile = Profile(cli_ctx=cli_ctx)
     credential, _, _ = profile.get_login_credentials(resource='https://managedhsm.azure.net')
```

Comment:
If keyvault data plane SDK can handle challenge, hard-coding `resource` is unnecessary.

Reply:
`get_raw_token` invocation should be dropped after fully migrating to Track 2,

https://github.com/Azure/azure-cli/blob/5ee89a457012288ad1e542d706f18bc1ae736f0e/src/azure-cli/azure/cli/command_modules/keyvault/_client_factory.py#L156-L157

Reply:
Removed `resource='https://managedhsm.azure.net'`. Sure, track1 client factory will be deleted after migration. `get_raw_token` will be dropped by then.